compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -3539,6 +3543,32 @@ public BLangNode transform(ModuleXMLNamespaceDeclarationNode xmlnsDeclNode) {
         return xmlns;
     }
 
+    @Override
+    public BLangNode transform(ClientDeclarationNode clientDeclarationNode) {
+        BLangClientDeclaration clientDeclaration = (BLangClientDeclaration) TreeBuilder.createClientDeclarationNode();
+
+        clientDeclaration.uri = createSimpleLiteral(clientDeclarationNode.clientUri());
+        clientDeclaration.prefix = createIdentifier(clientDeclarationNode.clientPrefix());
+        clientDeclaration.pos = getPosition(clientDeclarationNode);
+        clientDeclaration.annAttachments = applyAll(clientDeclarationNode.annotations());
```

Comment:
We can extract the common part. BTW do we need two bLang nodes for this? `XMLNamespaceDeclarationNode` has used two nodes anyway

Reply:
Will extract the common logic out in https://github.com/ballerina-platform/ballerina-lang/pull/37471. We need two nodes because one is a statement and other is not.